apps/roboshield/src/components/BlockRenderer/BlockRenderer.tsx

```diff
@@ -0,0 +1,29 @@
+import PageHeader from "@/roboshield/components/PageHeader/PageHeader";
+import { Page } from "@/root/payload-types";
+
+type BlockType = Page["blocks"];
+interface BlockRendererProps {
+  blocks: BlockType;
+}
+
+const components = {
+  "page-header": PageHeader,
+};
+
+type BlockSlug = keyof typeof components;
```

Comment:
Isn't `BlockSlug` always the same as `BlockType` (or should be the same)?
```ts
type BlockSlug = BlockType
```

And isn't `BlockRendererProps` just a subset of `Page`?
```ts
interface BlockRendererProps extends Pick<Page, "blocks"> {
}
```